source/UniverseObjects.cpp

```diff
@@ -217,6 +223,10 @@ void UniverseObjects::CheckReferences()
 	for(auto &&it : effects)
 		if(it.second.Name().empty())
 			NameAndWarn("effect", it);
+	// Variants are not serialized. Any changes via events are written as DataNodes and thus self-define.
+	for(auto &&it : variants)
+		if(!it.second.IsValid() && !deferred["variant"].count(it.first))
```

Comment:
This should check whether the variant is unnamed, not whether it is invalid. Or else the warning doesn't make sense.

Reply:
This is copied from Fleet's behavior just below this. If this needed changed then would not Fleet also need changed?

Reply:
Oh yeah I can see why Fleet (and Variant) use IsValid now. I guess it does make sense for them. sorry